### PR TITLE
Switch from Gson to kotlinx.serialization

### LINKED ISSUE #28: Switch from Gson to kotlinx.serialization
We are currently using Gson for our deserialization needs. According to the [Github project](https://github.com/google/gson) it's now in maintenance mode and can be considered deprecated. Quote: 'Gson is currently in maintenance mode'.

I propse switching to the official serialization library kotlinx.serialization as a replacement.

Another alternative would be [Moshi](https://github.com/square/moshi), but I'd prefer going with the more "native" solution.

### PR DESCRIPTION
**Pull request by fanonwue**
Updated data models to be compatible with `kotlinx.serialization`, and update the internal HTTP client to be compatible as well.

Unlike my initial thoughts, it's NOT necessary to alter all internal API clients. The bug that I encountered is related to Fuel's extension package for `kotlinx.serialization`. It doesn't use the default serializers that are registered for collection classes, like List etc. This causes the serialization library to fallback to standard polymorphic serialization, which fails when trying to deserialize JSON arrays.

The workaround is pretty simple: Just pass in the default serializers from the default serializers module.

Still need to either update some of the test response files or adjust the data models, since the tests fail due to missing required fields. Need to check the documentation for that.

This fixes #28

### CODE REVIEW

**Comment by Kryszak** [comment]
> Still need to either update some of the test response files or adjust the data models, since the tests fail due to missing required fields

Are the errors present because of errors in examples from docs, or due to our data models? I'd say we should first check our models, maybe we lack nullable types here and there. After that I'd check if the response models changed, some of the ones used in tests are 4 years old.

**Comment by Kryszak** [comment]
> Wouldn't have catched this without this new "strictness".

Nice to have another double-check thing to watch our back.

File: build.gradle.kts [hunk 1]
```diff
@@ -3,10 +3,12 @@ plugins {
     jacoco
     id("maven-publish")
     kotlin("jvm") version "1.8.20"
+    kotlin("plugin.serialization") version "1.8.20"
 }
 
 val fuelVersion = "2.3.1"
 val gsonVersion = "2.10.1"
```
Comment (Kryszak) [inline build.gradle.kts:10]: GSON version is no longer needed

Comment (fanonwue) [reply build.gradle.kts]: It's removed now.

**Comment by fanonwue** [comment]
I think I've done the bulk of the work now. Turns out there were quite a few models that were incomplete.

I've switched nullable collections of optional properties to have a default collection by default instead of being null, but if you want to keep them as null I'd switch them back. I just like the idea of them not being null, because the API is sometimes inconsistent in that it returns empty lists for some properties, while leaving out others completely (which are null, then).

I will also have to add some tests for the new subtypes I have introduced for some of the data models.

**Comment by Kryszak** [comment]
For the nullable collections we can keep them as empty collections no problem.

**Review by Kryszak — APPROVED** [review]

**Pull request merged**